### PR TITLE
Fix ItineraryBody layout

### LINKED ISSUE #222: Route destination and transit operator name not flush
Some route destinations and operator names are not rendered flush with the baseline as shown in the screenshot below:

image

### LINKED ISSUE #394: ItineraryBody layout issues
The `ItineraryBody` component has some layout issues circled in the screen excerpts below:
- Words for the walk/bike description and transit route names should be on the same baseline
- Long walk/bike descriptions and long transit route names should be aligned left with less spacing between lines (see https://newplanner.trimet.org/map/#/ for the original layout)

image

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Fix #222, fix #394 (see screenshot below).

image

### CODE REVIEW

File: packages/itinerary-body/src/styled.tsx [hunk 1]
```diff
@@ -522,15 +533,21 @@ export const StepDescriptionContainer = styled.div`
   margin-left: 24px;
   line-height: 1.25em;
   padding-top: 1px;
+
+  & > span {
+    margin-right: 1ch;
+  }
+`;
+
+export const StepsHeaderAndMap = styled.span`
```
Comment (miles-grant-ibigroup) [inline packages/itinerary-body/src/styled.tsx:542]: Would it be possible to rename this? This makes it seem like a map is being rendered

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking really good! Thanks for the cleanup

File: packages/itinerary-body/src/styled.tsx [hunk 2]
```diff
@@ -280,19 +281,17 @@ export const LegBody = styled.div`
...
-
-  > div {
-    display: table-cell;
-  }
+// Use <span> for correct semantics as it is the contents of a button or a link.
+export const LegDescription = styled.span`
+  align-items: center;
+  display: flex;
+  line-height: 16px;
+  min-height: 31px;
```
[the remark this answers is not in the document]
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:294]: I think the intent for that was for this portion to have the same height whether single-line or multi-line text, but the initial work did not account for multi-line text.

Also, these heights are based on the height of the Hind font at the font sizes used.

File: packages/itinerary-body/src/styled.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:542]: This `styled` module is exported at package level, so renaming would be a breaking change.

Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx:542]: > This `styled` module is exported at package level, so renaming would be a breaking change.

Disregard response above.

Comment (binh-dam-ibigroup) [reply packages/itinerary-body/src/styled.tsx]: Yes I didn't finish typing the name! Renamed in https://github.com/opentripplanner/otp-ui/pull/395/commits/5ad35a818775c148fb1ce4a7f52c3bb627feb22c.

**Review by philip-cline — APPROVED** [review]
Looks good to me!

**Pull request merged**